Author: Ethan Hsieh <ethan.hsieh@canonical.com>
Description: Fix bad CRC issue
 Some of the config options defined in the original Ubuntu nitrogen6x patch
 are not yet migrated to Kconfig as of our current u-boot version. As a
 workaround, define those the old-fashoned way.
--- a/configs/nitrogen6q2g_defconfig
+++ b/configs/nitrogen6q2g_defconfig
@@ -40,7 +40,6 @@
 CONFIG_ENV_FAT_INTERFACE="mmc"
 CONFIG_ENV_FAT_DEVICE_AND_PART="0:auto"
 CONFIG_ENV_FAT_FILE="uboot.env"
-CONFIG_ENV_SIZE=0x20000
 CONFIG_DWC_AHSATA=y
 CONFIG_USB_FUNCTION_FASTBOOT=y
 CONFIG_FASTBOOT_BUF_ADDR=0x12000000
@@ -72,4 +71,3 @@
 CONFIG_VIDEO=y
 # CONFIG_VIDEO_SW_CURSOR is not set
 CONFIG_OF_LIBFDT=y
-CONFIG_SYS_REDUNDAND_ENVIRONMENT=y
--- a/include/configs/nitrogen6x.h
+++ b/include/configs/nitrogen6x.h
@@ -11,6 +11,10 @@
 
 #include "mx6_common.h"
 
+#if defined(CONFIG_TARGET_NITROGEN6X)
+#define CONFIG_SYS_REDUNDAND_ENVIRONMENT
+#endif
+
 #define CONFIG_MACH_TYPE	3769
 
 /* Size of malloc() pool */
@@ -149,7 +153,9 @@
 	(CONFIG_SYS_INIT_RAM_ADDR + CONFIG_SYS_INIT_SP_OFFSET)
 
 /* Environment organization */
-#ifndef CONFIG_ENV_SIZE
+#if defined(CONFIG_TARGET_NITROGEN6X)
+#define CONFIG_ENV_SIZE			SZ_128K
+#else
 #define CONFIG_ENV_SIZE			(8 * 1024)
 #endif
 
